src/wallet.rs

```diff
@@ -125,41 +130,33 @@ pub fn parse_command(
     Ok(response)
 }
 
-pub fn process_command(
-    config: &WalletConfig,
-    client: &mut ThinClient,
-) -> Result<String, Box<error::Error>> {
+pub fn process_command(config: &WalletConfig) -> Result<String, Box<error::Error>> {
     match config.command {
-        // Check client balance
+        // Get address of this client
         WalletCommand::Address => Ok(format!("{}", config.id.pubkey())),
-        WalletCommand::Balance => {
-            println!("Balance requested...");
-            let balance = client.poll_get_balance(&config.id.pubkey());
-            match balance {
-                Ok(balance) => Ok(format!("Your balance is: {:?}", balance)),
-                Err(ref e) if e.kind() == ErrorKind::Other => {
-                    Ok("No account found! Request an airdrop to get started.".to_string())
-                }
-                Err(error) => Err(error)?,
-            }
-        }
         // Request an airdrop from Solana Drone;
         // Request amount is set in request_airdrop function
         WalletCommand::AirDrop(tokens) => {
             println!(
                 "Requesting airdrop of {:?} tokens from {}",
                 tokens, config.drone_addr
             );
-            let previous_balance = client.poll_get_balance(&config.id.pubkey()).unwrap_or(0);
+            let params = format!("[\"{}\"]", config.id.pubkey());
+            let previous_balance = WalletRpcRequest::GetBalance
+                .make_rpc_request(&config.rpc_addr, 1, Some(params))?
+                .as_i64()
+                .unwrap_or(0);
```

Comment:
shouldn't we error out to the user if GetBalance fails?   Might have been a network connection issue

Reply:
Well, if `make_rpc_request` fails, it will return whatever error and not make it to the `unwrap_or`. But checking the return type on this first call is still wise, so I'm updating.